demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <utils/images_capture.h>
+#include <opencv2/gapi.hpp>
+#include <chrono>
+
+namespace custom {
+class CustomCapSource : public cv::gapi::wip::IStreamSource
+{
+public:
+    explicit CustomCapSource(std::shared_ptr<ImagesCapture>& cap, const int batch_size, const int batch_fps)
+    : cap(cap), batch_size(batch_size), batch_fps(batch_fps) {
+        prep();
+    }
+
+protected:
+    std::shared_ptr<ImagesCapture> cap;
+    int batch_size = 0; // size of batch
+    int batch_fps = 0;  // constant FPS for batch
+    cv::Mat batch;
+    int first_el = 0;   // number of first image of batch
+    int counter = 0;    // amount of images in batch
+    std::chrono::steady_clock::time_point start_time; // time from previous read()
+    bool first_pulled = false; // is first already pulled
+    void prep() {
+        GAPI_Assert(batch.empty());
+        cv::Mat frame = cap->read();
+        if (!frame.data) {
+            GAPI_Assert(false && "Couldn't grab the frame");
+        }
```

Comment:
What if the input stream is empty? This may be valid situation and the pipeline won't produce anything in this case (the very first `pull()` returns `false`).